tgui/packages/tgui/interfaces/SeedFabricator.js

```diff
@@ -0,0 +1,152 @@
+
+const DefaultSort = {
+  Fruit: 1,
+  Vegetable: 2,
+  Herb: 3,
+  Flower: 4,
+  Miscellaneous: 5,
+  Other: 6,
+};
+
+export const SeedFabricator = (props, context) => {
+  const { data } = useBackend(context);
+  const { canVend, isWorking, maxSeed, name, seedCount } = data;
+  const categories = data.seedCategories || [];
+
+  categories.sort((a, b) => ((DefaultSort[a.name] || DefaultSort.Other) - (DefaultSort[b.name] || DefaultSort.Other)));
+
+  const [dispenseAmount, setDispenseAmount] = useLocalState(context, 'dispenseAmount', 1);
+
+  return (
+    <Window
+      title={name}
+      width={500}
+      height={600}>
+      <Window.Content>
+        {!isWorking && (
+          <Modal textAlign="center"
+            width={35}
+            height={10}
+            fontSize={3}
+            fontFamily="Courier"
+            color="red">
+            <Blink time={500}>
+              <Icon name="exclamation-triangle" pr={1.5} />
+              MALFUNCTION
+              <Icon name="exclamation-triangle" pl={1.5} />
+            </Blink>
+            CHECK WIRES
+          </Modal>
+        )}
+        <Section>
+          <Flex>
+            <Flex.Item bold pr={1}>
+              Dispense:
+            </Flex.Item>
+            <Flex.Item basis={6} grow>
+              <NumberInput
+                value={dispenseAmount}
+                format={value => value + pluralize(" seed", value)}
+                minValue={1}
+                maxValue={10}
+                onDrag={(e, value) => setDispenseAmount(value)}
+              />
+            </Flex.Item>
+            <Flex.Item grow={2}>
+              <ProgressBar
+                value={Math.max(0, maxSeed - seedCount)}
+                maxValue={maxSeed}
+                ranges={{
+                  yellow: [5, Infinity],
+                  bad: [-Infinity, 5],
+                }}
+              >
+                <Icon name="bolt" />
+              </ProgressBar>
+            </Flex.Item>
+          </Flex>
+        </Section>
+        <Section>
+          {!canVend && (
+            <Modal textAlign="center"
+              width={25}
+              height={5}
+              fontSize={2}
+              fontFamily="Courier"
+              color="yellow">
+              <Blink interval={500} time={500}>
+                <Icon name="bolt" pr={1.5} />
+              </Blink>
+              UNIT RECHARGING
+            </Modal>
+          )}
+          {categories.map((category, index) => (
+            <SeedCategory
+              key={category.name}
+              category={category}
+              dispenseAmount={dispenseAmount} />
+          ))}
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const seedsSorter = (a, b) => a.name.localeCompare(b.name);
+
+const SeedCategory = (props, context) => {
+  const { act } = useBackend(context);
+  const { category, dispenseAmount } = props;
+  const { name, seeds } = category;
```

Comment:
Not requiring, just as an FYI that you can write as:
```suggestion
  const {
    category: {
      name,
      seeds,
    },
    dispenseAmount,
  } = props;
```